include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp

```diff
@@ -1465,15 +1465,23 @@ template<int dim, typename Number>
 void
 SpatialOperatorBase<dim, Number>::move_grid(double const & time) const
 {
-  grid_motion->update(time, false);
+  // Driver::ale_update() is responsible for the preconditioner update
+  grid_motion->update(
+    time,
+    false /* print_solver_info */,
+    dealii::numbers::invalid_unsigned_int /* time_step_number used for preconditioner update */);
```

Comment:
The `move_grid()` and `move_grid_and_update_dependent_data_structures()` are used during initialization/postprocessing as far as I can tell?
`grep -rnw -e 'move_grid'` in `exadg/include` gives
"
exadg/convection_diffusion/spatial_discretization/interface.h:108:  move_grid(double const & time) const = 0;
exadg/convection_diffusion/spatial_discretization/operator.h:202:  move_grid(double const & time) const;
exadg/convection_diffusion/spatial_discretization/operator.cpp:821:Operator<dim, Number>::move_grid(double const & time) const
exadg/convection_diffusion/time_integration/time_int_bdf.cpp:72:    pde_operator->move_grid(this->get_time());
exadg/convection_diffusion/time_integration/time_int_bdf.cpp:80:        pde_operator->move_grid(this->get_previous_time(i));
exadg/convection_diffusion/time_integration/time_int_bdf.cpp:141:    pde_operator->move_grid(this->get_time());
exadg/convection_diffusion/time_integration/time_int_bdf.cpp:154:      pde_operator->move_grid(this->get_previous_time(i));
exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp:1466:SpatialOperatorBase<dim, Number>::move_grid(double const & time) const
exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.h:389:  move_grid(double const & time) const;
exadg/incompressible_navier_stokes/time_integration/time_int_bdf_coupled_solver.cpp:67:    pde_operator->move_grid(this->get_time());
exadg/incompressible_navier_stokes/time_integration/time_int_bdf_coupled_solver.cpp:82:      pde_operator->move_grid(this->get_previous_time(i));
exadg/incompressible_navier_stokes/time_integration/time_int_bdf_pressure_correction.cpp:147:    pde_operator->move_grid(this->get_time());
exadg/incompressible_navier_stokes/time_integration/time_int_bdf_pressure_correction.cpp:160:      pde_operator->move_grid(this->get_previous_time(i));
exadg/incompressible_navier_stokes/time_integration/time_int_bdf.cpp:99:    operator_base->move_grid(this->get_time());
exadg/incompressible_navier_stokes/time_integration/time_int_bdf.cpp:107:        operator_base->move_grid(this->get_previous_time(i));
exadg/incompressible_navier_stokes/time_integration/time_int_bdf_dual_splitting.cpp:139:    pde_operator->move_grid(this->get_time());
exadg/incompressible_navier_stokes/time_integration/time_int_bdf_dual_splitting.cpp:152:      pde_operator->move_grid(this->get_previous_time(i));

"

So I would rather stick to the convention of triggering the update only in `Driver::ale_update()`.

Reply:
see comment https://github.com/exadg/exadg/pull/404/files#r1177448382